macaroonbakery/httpbakery/agent/agent.py

```diff
@@ -123,26 +127,22 @@ def legacy_interact(self, client, location, visit_url):
         '''Implement LegacyInteractor.legacy_interact by obtaining
         the discharge macaroon using the client's private key
         '''
+        log.info('agent legacy_interact, visit_url {}'.format(visit_url))
         agent = self._find_agent(location)
-        pk_encoded = self._auth_info.key.public_key.encode().decode('utf-8')
-        value = {
-            'username': agent.username,
-            'public_key': pk_encoded,
-        }
-        # TODO(rogpeppe) use client passed into interact method.
-        client = httpbakery.Client(key=self._auth_info.key)
-        client.cookies.set_cookie(utils.cookie(
+        client = copy.copy(client)
```

Comment:
Why do you need a copy ?

Reply:
Yes, it would be nice to have a comment about that. Also, do we need deepcopy?

Reply:
I've added a comment. Deep copy isn't necessary, as all we want to do is avoid side-effecting the client by setting its key field.